### PR TITLE
feat: Upload Default CDOGS Template Frontend

### PR DESCRIPTION
**Pull request by vijaivir**
# Description
This PR introduces new functionality that allows users to provide a pre-existing CDOGS template that form submitters can use in the printing dialog. This complements #1317 and provides the front-end logic for the feature. 

A new 'CDOGS Template' dropdown has been added in the 'Manage Form' page where users may manage a document template associated with that form. Users can upload and delete the document template.  

Screenshot 2024-04-22 at 12 32 36 PM

This was a highly requested feature on [Fider](https://chefs-fider.apps.silver.devops.gov.bc.ca/posts/140/one-click-cdogs-print). 

[Jira Link ](https://bcdevex.atlassian.net/jira/software/c/projects/FORMS/boards/95?selectedIssue=FORMS-1110&sprints=749)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/frontend/src/internationalization/trans/chefs/en/en.json [hunk 1]
```diff
@@ -64,7 +64,22 @@
     "status": "Status",
     "update": "Update",
     "cancel": "Cancel",
-    "eventSubscription": "Event Subscription"
+    "eventSubscription": "Event Subscription",
+    "cdogsTemplate": "CDOGS Template"
```
Comment (WalterMoar) [inline app/frontend/src/internationalization/trans/chefs/en/en.json:68]: We'll also need the translations for these, yes?

File: app/frontend/src/components/forms/manage/DocumentTemplate.vue [hunk 2]
```diff
@@ -0,0 +1,372 @@
+<script>
```
Comment (WalterMoar) [inline app/frontend/src/components/forms/manage/DocumentTemplate.vue:1]: I think @jasonchung1871 and/or @nimya-aot would be able to help with how to write tests for this. I'm not sure what the goal/approach is for testing the frontend.

**Review by WalterMoar — COMMENTED** [review]
Looks good to me on how the API is used and functionally it seems great! @jasonchung1871 knows more about the frontend design and test coverage and that sort of thing, so I'll leave it to him for final approval.

**Comment by jasonchung1871** [comment]
> Looks good to me on how the API is used and functionally it seems great! @jasonchung1871 knows more about the frontend design and test coverage and that sort of thing, so I'll leave it to him for final approval.

It does look good. Ideally, we could get coverage on all of the new front end script. Not necessarily 100% coverage but just the new content. It's good as it is though.

**Review by jasonchung1871 — APPROVED** [review]

**Pull request merged**